internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -125,46 +165,209 @@ func GetClusterMap() ([]byte, *azcore.ETag, error) {
+			Name: path,
+		})
+		if err != nil {
+			log.Err("updateHandleCount :: Failed to get handle count for %s : %v", path, err)
+			return -1, err
+		}
+
+		// We never create file metadata blob w/o opencount property set.
+		if attr.Metadata["opencount"] == nil {
+			log.Err("updateHandleCount :: File metadata blob found w/o opencount property: %s", path)
+			return -1, fmt.Errorf("Opencount property not found in metadata for path %s. Issue needs to be debugged.", path)
+		}
+		openCount, err = strconv.Atoi(*attr.Metadata["opencount"])
+		if err != nil {
+			log.Err("GetFileOpenCount :: Failed to parse handle count for path %s : %v", path, err)
+			return -1, err
+		}
+		if increment {
+			openCount++
+		} else {
+			openCount--
+		}
+		if openCount < 0 {
+			log.Err("updateHandleCount :: Handle count is negative for path %s : %d", path, openCount)
+			return -1, fmt.Errorf("Handle count is negative for path %s : %d", path, openCount)
+		}
+		openCountStr := strconv.Itoa(openCount)
+		attr.Metadata["opencount"] = &openCountStr
+
+		// Set the new metadata in storage
+		err = m.storageCallback.SetMetaPropertiesInStorage(internal.SetMetadataOptions{
+			Path:      filepath.Join(m.mdRoot, path),
+			Metadata:  attr.Metadata,
+			Etag:      to.Ptr(azcore.ETag(attr.ETag)),
+			Overwrite: true,
+		})
+		if err != nil {
+			if bloberror.HasCode(err, bloberror.ConditionNotMet) {
+				log.Warn("updateHandleCount :: SetPropertiesInStorage failed for path %s due to ETag mismatch, retrying...", path)
+
+				// Apply exponential backoff
+				log.Debug("updateHandleCount :: Retrying in %d milliseconds...", backoff)
+				time.Sleep(backoff)
+
+				// Double the backoff time, but cap it at maxBackoff
+				backoff *= 2
+				if backoff > maxBackoff {
+					backoff = maxBackoff
+				}
+
+				// Check if retrying has exceeded a minute
+				if time.Since(retryTime) >= time.Minute {
+					log.Warn("updateHandleCount :: Retrying exceeded one minute for path %s, exiting...", path)
+					return -1, fmt.Errorf("Retrying exceeded one minute for path %s", path)
+				}
+				continue
+			} else {
+				log.Err("updateHandleCount :: Failed to update metadata property for path %s : %v", path, err)
+				return -1, err
+			}
+		} else {
+			break
+		}
+	}
+	return int64(openCount), nil
 }
 
 // GetFileOpenCount returns the current open count for a file
 func (m *BlobMetadataManager) getFileOpenCount(filePath string) (int64, error) {
-	// Dummy implementation
-	return 0, nil
+	path := filepath.Join(m.mdRoot, "Objects", filePath)
+	prop, err := m.storageCallback.GetPropertiesFromStorage(internal.GetAttrOptions{
+		Name: path,
+	})
+	if err != nil {
+		log.Err("GetFileOpenCount :: Failed to get handle count for path %s : %v", path, err)
+		return -1, err
+	}
+	openCount, ok := prop.Metadata["opencount"]
+	if !ok {
+		log.Err("GetFileOpenCount :: openCount not found in metadata for path %s : Error %v", path, err)
+		// TODO :: Add asserts
```

Comment:
common.Assert(false, fmt.Sprintf("openCount not found in metadata for path %s", path))